### ISSUE TITLE
Pattern matching 

### ISSUE THREAD

**Issue by danielo515** [flywheelsports/hydra#198]
Hello,
I'm doing a deep research of all the microservices frameworks out there. I've checked Seneca, Hemera, Moleculer and now hydra.

Something I really like from Hemera and Seneca is pattern matching. This feature really allows incrementally adding features. As a bonus there is absolutely no need to know which service can handle the message beforehand, super cool.

Hydra has another thing that I also like a loot, which is direct service communications. There are sometimes when you need to transfer a big payload and know who should handle it, in that case direct http or tcp communication seems better. (if hydra is not doing this please clarify it, thanks) 

My question is, are you aware of pattern matching and if so, do you have plans to implement it? 

Regards

**Comment by danielo515** [comment]
We are taking Hydra in consideration for starting a new project, however this feature is important to us and I would like to have it clarified.

Regards

**Comment by danielo515** [comment]
Does not support, and will never support I guess ?

**Comment by cjus** [comment]
@danielo515 we're always up for a PR!  Even a write-up of what others are doing would be a great way to kick this off! That way we can at least start discussion until someone picks up the feature.

**Comment by danielo515** [comment]
Hello.
Do you mean a writeup on this same issue or something more like an article or blog post?

**Comment by cjus** [comment]
I was thinking more of a proposal document. Once which lists requirements or features.  A doc which looks at what others are doing and says "it would be cool if hydra did this..."  The document doesn't have to be long and can, in fact, be opinionated.  Such a document would help to start a conversation.

**Comment by cjus** [comment]
@danielo515 BTW, have you read Richard Rodger's book "The Tao of Microservices"?

**Comment by danielo515** [comment]
HEllo @cjus 

> The Tao of Microservices

No I did not, but I think I will. I am reading buliding microservices with node.js, which uses seneca. Maybe I could read that one after. Thanks for your suggestions.

**Comment by sjmcdowall** [comment]
@danielo515 -- The author of The Tao of Microservices is the primary author of Seneca :)

**Comment by danielo515** [comment]
@sjmcdowall
Absolutely. I read several blogs post of the Seneca author (which I'm unable to find again :sad:). He has very good ideas and a very good vision about how to make something scalable and extensible. I will read the book for sure
